src/libostree/ostree-repo-prune.c

```diff
@@ -401,6 +417,11 @@ ostree_repo_prune_from_reachable (OstreeRepo        *self,
                                   GCancellable      *cancellable,
                                   GError           **error)
 {
+  g_autoptr(OstreeRepoAutoLock) lock = NULL;
+  lock = ostree_repo_auto_lock_push (self, TRUE, cancellable, error);
+  if (!lock)
+    return FALSE;
```

Comment:
One thought: if you changed the definition of `ostree_repo_auto_lock_push()` to:
`gboolean ostree_repo_auto_lock_push (OstreeRepo *self, OstreeRepoLockFlags flags, OstreeRepoAutoLock **out_lock, GCancellable *cancellable, GError **error)`, then all the uses of it could simplify to:
```
g_autoptr(OstreeRepoAutoLock) lock = NULL;
if (!ostree_repo_auto_lock_push (self, OSTREE_REPO_LOCK_EXCLUSIVE, &lock, cancellable, error))
  return FALSE;
```

Reply:
I could go either way on this and did consider that interface. I do generally like ostree's boolean return values with supplemental information in out params. In this case it seemed overkill to me, though, as both the return value and the out param would be transmitting the same information. Would the out param be a requirement? it would be useless to use the function otherwise. Not that we're counting cycles in ostree, but it's also less efficient to put another parameter on the stack and fill it in.

You can, of course, do the call and check in one line if you want with the current interface:
```
if (!(lock = ostree_repo_auto_lock_push(...)))
  return FALSE;
```
That's definitely not as aesthetically pleasing as the interface you've suggested, which is also why I prefer to write it on 2 lines.